domain/model/state/state.go

```diff
@@ -566,17 +619,79 @@ AND owner_name = ?
 		)
 	}
 
-	if !credentialName.Valid {
-		return cloudName, credential.Key{}, nil
+	return cloudName, credentialKey, nil
+}
+
+// NamespaceForModel returns the database namespace that is provisioned for a
+// model id. If no model is found for the given id then a [modelerrors.NotFound]
+// error is returned. If no namespace has been provisioned for the model then a
+// [modelerrors.ModelNamespaceNotFound] error is returned.
+func (st *State) NamespaceForModel(ctx context.Context, id coremodel.UUID) (string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	q := `
+    SELECT m.uuid, mn.namespace
```

Comment:
Formatting here is different, uses indentation. Compare with existsStmt in GetFinaliser()